userguide/sound_classifier/how-it-works.md

```diff
@@ -0,0 +1,52 @@
+# How Does this Work?
+
+Training and making predctions, for a sound classifier model, is a three
+stage process. The first stage does signal preprocessing. In
+the second step we use a pretrained neural network to extract deep
+features. Finally in the last step, we use a custom neural network.
+
+## Signal Preprocessing Pipeline Stage
+Several audio transformation happen during this stage. For someone without
+sound domain knowledge, this is probably the most complicated part.
+Nothing about this stage is updated based on the input data (i.e. nothing
+is learned in this stage).
+
+At a high level the preprocessing pipeline does the following:
+* The raw pulse code modulation data from the wav file is converted to
+floats on a [-1.0, +1.0] scale.
+* The data is resampled to only 16,000 sample per second.
+* The data is broken up in to several overlapping windows.
+* A [Hamming Window](https://en.wikipedia.org/wiki/Window_function#Hann_and_Hamming_windows) is applied to each windows.
+* The [Power Spectrum](https://en.wikipedia.org/wiki/Spectral_density#Power_spectral_density) is calculated, using a [Fast Fourier Transformation](https://en.wikipedia.org/wiki/Fourier_transform).
+* Frequencies above and below certain thresholds are dropped.
+* Mel Frequency Filter Banks are applied.
+* Finally the log is taken of all values.
+
+The preprocessing pipeline takes 975ms worth of audio as input (exact
+input lenght depends on sample rate) and produce an array of shape
+(96, 64).
+
+## VGGish Feature Extraction Stage
+VGGish is a pretrained [Convolutional Neural Network](https://en.wikipedia.org/wiki/Convolutional_neural_network) from Google,
+see [their paper](https://ai.google/research/pubs/pub45611) and [their GitHub page](https://github.com/tensorflow/models/tree/master/research/audioset) for more details. As the name suggests, the architecture of
+this network is inspired by the famous VGG networks used for image
+classification. The network consists of a series of convolution and
+activation layers, optionally followed by a max pooling layer.
+This network contains 17 layers in total.
+
+This network is kept static during model training. We have removed the
+last three layers of the original VGGish model. We use the widest
+layer, from the original network, as our input data for the final
+stage. This modified VGGish model ouputs a double vector of length
+12,288. On non-Linux systems, the model has also been eight bit
+quantized, to reduce its size.
+
+## Custom Neural Network Stage
+This is the only stage which is updated based on the input data.
+During training of a sound classifier this model is trained using
+the features from VGGish and the input labels. During prediction a
```

Comment:
"During prediction a only a forward pass is done" -> "During prediction, only a forward pass is done"

Reply:
Done.